content/docs/5.1/develop/rawkv/cas.md

```diff
@@ -0,0 +1,66 @@
+---
+title: CAS
+description: How to use RawKV's CAS API 
+menu:
+    "5.1":
+        parent: RawKV
+        weight: 3
+---
+
+This document walks you through how to use RawKV’s `CAS (Compare And Swap)` API.
+
+In RawKV, compare-and-swap (CAS) is an atomic instruction used in multithreading to achieve synchronization, which is the atomic equivalent of:
+
+```
+if get(key) == old_value {
+	put(key, new_value);
+	return true;
+}
+return false;
```

Comment:
```suggestion
let prev_value = get(key);
if prev_value == request.prev_value {
	put(key, request.value);
}
return prev_value;
```

Reply:
the example is taken from https://golang.org/pkg/sync/atomic/

Reply:
However, it is compare and set.

Reply:
for pseudo-code, it's better to use c/c++/java since they are widely used by users.

Reply:
What about removing 'let' in the suggestion?

Reply:
> What about removing 'let' in the suggestion?

I'm ok with that.

Reply:
> However, it is compare and set. Tikv is compare and swap.

Other language's examples will also be listed here latter.

Reply:
fixed